### ISSUE TITLE
Not checking reference-style links

### ISSUE THREAD

**Issue by dbanty** [HiDeoo/starlight-links-validator#14]
### Describe the bug

If you use [reference style links](https://www.markdownguide.org/basic-syntax/#reference-style-links), this check does not fail when the relative path does not exist.

```markdown

[this will be caught](/link-that-does-not-exist)

[but this will not]

[but this will not]: /link-that-does-not-exist
```

### To Reproduce

I created [this repo](https://github.com/dbanty/reproduce-link-checker-reference) from the starlight starter and replaced the content with a single reference that should be caught and is not.

### Expected behavior

Reference-style relative links should be caught the same way as inline links.

### How often does this bug happen?

Every time

**Comment by HiDeoo** [comment]
Thanks for the detailed issue 🙌 

This should now be fixed in [v0.4.2](https://github.com/HiDeoo/starlight-links-validator/releases/tag/v0.4.2).

**Comment by dbanty** [comment]
Wow, that was fast, thanks so much! It works, and my site now has several fewer broken links 😁
